build-tools/packages/build-cli/src/commands/release.ts

```diff
@@ -0,0 +1,834 @@
+                            );
+                            if (!context.originalBranchName.startsWith("release/")) {
+                                this.warn(
+                                    `Patch release should only be done on 'release/*' branches, but current branch is '${context.originalBranchName}'.\nYou can skip this check with --no-branchCheck.'`,
+                                );
+                                this.machine.action("failure");
+                                this.exit();
+                            }
+
+                            break;
+                        }
+
+                        case "major":
+                        case "minor": {
+                            this.verbose(
+                                `Checking if ${context.originalBranchName} is 'main', 'next', or 'lts'.`,
+                            );
+                            if (!["main", "next", "lts"].includes(context.originalBranchName)) {
+                                this.warn(
+                                    `Release prep should only be done on 'main', 'next', or 'lts' branches, but current branch is '${context.originalBranchName}'.`,
+                                );
+                                this.machine.action("failure");
+                                this.exit();
+                            }
+                        }
+                    }
+                } else {
+                    this.warn(
+                        `Not checking if current branch is a release branch: ${context.originalBranchName}`,
+                    );
+                }
+
+                this.machine.action("success");
+                break;
+            }
+
+            case "CheckInstallBuildTools": {
+                const installQuestion: inquirer.ConfirmQuestion = {
+                    type: "confirm",
+                    name: "install",
+                    message: `Do you want to install the Fluid build-tools? You don't need to do this if you installed them globally.`,
+                };
+
+                const answer = await inquirer.prompt(installQuestion);
+                if (answer.install === true) {
+                    this.log(`Installing build-tools so we can run build:genver`);
+                    const buildToolsMonoRepo = context.repo.releaseGroups.get(
+                        MonoRepoKind.BuildTools,
+                    )!;
+                    const ret = await buildToolsMonoRepo.install();
+                    if (ret.error) {
+                        this.errorLog("Install failed.");
+                        this.machine.action("failure");
+                    }
+                } else {
+                    this.warn(`Skipping installation.`);
+                }
+
+                this.machine.action("success");
+                break;
+            }
+
+            case "CheckMainNextIntegrated": {
+                // TODO: Implement this
```

Comment:
Is this still relevant?

Reply:
Yeah, only used in major releases, so coming in a future PR.